view/frontend/web/js/view/payment/adyen-methods.js

```diff
@@ -75,6 +75,9 @@ define(
                         console.log('Fetching the payment methods failed!');
                     });
                 };
+                if (quote.shippingAddress() === null) {
+                    retrievePaymentMethods();
```

Comment:
Should retrievePaymentMethods be changed then to use billingAddress as well? How does retrievePaymentMethods get the shippingAddress if it is null?

Reply:
At this point for the virtual product there are not shipping information so we could retrieve the pm if it is null. The billing address is for a later step